xbmc/guilib/GUIVisualisationControl.cpp

```diff
@@ -298,21 +298,22 @@ void CGUIVisualisationControl::UpdateTrack()
   std::string albumArtist(tag->GetAlbumArtistString());
   std::string genre(StringUtils::Join(tag->GetGenre(), CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_musicItemSeparator));
 
-  VisTrack track = {0};
-  track.title       = tag->GetTitle().c_str();
-  track.artist      = artist.c_str();
-  track.album       = tag->GetAlbum().c_str();
-  track.albumArtist = albumArtist.c_str();
-  track.genre       = genre.c_str();
-  track.comment     = tag->GetComment().c_str();
-  track.lyrics      = tag->GetLyrics().c_str();
-  track.trackNumber = tag->GetTrackNumber();
-  track.discNumber  = tag->GetDiscNumber();
-  track.duration    = tag->GetDuration();
-  track.year        = tag->GetYear();
-  track.rating      = tag->GetUserrating();
-
-  m_instance->OnAction(VIS_ACTION_UPDATE_TRACK, &track);
+  VIS_TRACK* track = new VIS_TRACK(); // allocate by his size
```

Comment:
Can you be more descriptive about why this is allocated?

Reply:
Ah yeah, is about this place https://github.com/xbmc/xbmc/pull/18251/files#diff-3db4e9fc8c806bfb2460fe6b33d6b649R34

Maybe?:
- `Its size can exceed the available stack size, an explicit "new" is necessary.`

Reply:
And we can be a bit more specific about why it exceeds:

* `Its size can exceed the available stack size due to the 32K allocated for lyrics, so an explicit "new" is necessary.`

Reply:
@AlwinEsch after seeing the issue solved by https://github.com/xbmc/xbmc/pull/18264 I wonder if we shouldn't make the lyric size dynamic instead of moving the struct to the heap

Reply:
It's a bit the thing about safe "C" interface, but to have them allocated in matching size of lyrics text would be the best.

But not know how easily it can done, if they come only from Kodi and nowhere copied with the C++ part about on addon side. Then it can be a allocated pointer about.

Only becomes addon allowed to copy itself this class, where is done by 'CStructHdl` (https://github.com/xbmc/xbmc/blob/master/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h#L217-L282) and where makes a complete copy of "C" part for other place.

If for every value it becomes done by normal pointer instead of fixed array, must the code of related class do byself where can be much code text.

Can look to make one special alloc about them and the rest with "CStructHdl", hope that works.